include/xtypes/DynamicData.hpp

```diff
@@ -864,6 +866,196 @@ class DynamicData : public WritableDynamicDataRef
     /// \brief Request a writable reference from this DynamicData.
     /// \returns a WritableDynamicDataRef identifying th DynamicData.
     WritableDynamicDataRef ref() const { return WritableDynamicDataRef(*this); }
+
+    template<typename T, class = std::is_arithmetic<T>>
```

Comment:
Ok, I have several comments about it:
- Could you add an assert for primitive types?
- It is really need to use the switch?, Maybe a: `*static_cast<T*>(this->instance)` could be enough as you implement in last line of this PR.
- A design comment: I think that his feature is out from this scope. This is for me a soss utility, not a DynamicData utility. If an xtypes user want to cast a DynamicData. He can retrieve the value and cast it by himself.

Reply:
Comments addressed in person :).